docs/guide/setup-examples/remote_dev.md

```diff
@@ -0,0 +1,189 @@
+I recommend having the same project folder names
+on your server and on your laptop
+and keeping all your projects in the same folder.
+This will make advanced steps easier.
+
+Update the following keys:
+
+```json
+    // sftp, ftp or ftps
+    "type": "sftp",
+
+    "save_before_upload": true,
+    "upload_on_save": true,
+
+    "host": "server",
+    "user": "USER",
+    // Use full paths here, SFTP doesn't handle shortcuts like ~ or $HOME.
+    "remote_path": "/home/USER/github/YOUR_PROJECT",
+    "sftp_flags": ["-F", "/home/USER/.ssh/config"],
+```
+
+ssh to your server in a terminal.
+Open a file from your project, save it.
+The status bar should show SFTP uploading the file to your server.
+You now have a working setup.
+SFTP will automatically upload files on save.
+But if you modify files out of Sublime (eg git checkout, git pull),
+you will need to upload the whole folder: `SFTP: Upload Folder`
+
+Don't forget to buy an SFTP license if you find this setup useful.
+(I'm not affiliated in any way with SFTP plugin)
+
+[global gitignore]: https://stackoverflow.com/a/7335487/3561471
+
+# Adding rsync
+
+Uploading the whole folder is often a bit slow.
+Using `rsync` will be faster,
+it only pushes the changed files.
+SFTP should switch to use `rsync` at some point,
+but in the meantime,
+I use the following command:
+`rsync --verbose -rd --del --exclude='.git/' --filter=':- .gitignore'`
+Note that this command is _aggressive_,
+`--del` means that files on the server but not on your laptop will be deleted,
+unless they are in the `.gitignore`.
+The `--del` flag is useful if you rename a file locally.
+SFTP will upload the new version but won't remove the old one.
+Having `rsync --del` running from time to time allows to clean up this.
+Feel free to remove it
+if you have important files in your server project folder,
+that you don't want to risk to lose.
+
+I added this to my `~/.zshrc`
+
+```sh
+function rsync_git() {
+    if [[ ! -d $PWD/.git ]]; then
+        echo "Current dir $PWD isn't a valid git dir.
+        Please move to the root of a git dir."
+        return;
+    fi
+
+    target=${1:-server:\~/github}
+    repo=${2:-${PWD##*/}}
+    # Async is important to not block git checkout
+    echo "Async rsync of git repo $repo to $target/$repo ..."
+
+    rsync --verbose -rd --del --exclude='.git/' --filter=':- .gitignore' \
+        . "$target/$repo" &
+}
+
+export -f rsync_git
+```
+
+This allows me to run `rsync_git`
+when I'm at the root of my repo
+to upload the whole folder to my server.
+Since I want to have the code on my server
+to be always a mirror of the code on my laptop,
+I added git hooks to automatically run `rsync_git` when I checkout a branch:
+
+```sh
+echo "source ~/.zshrc; rsync_git" > .git/post-checkout
+```
+
+Note that it still takes a few seconds to run
+so if you start a job on your server just after pulling,
+you may end up running code from the old branch.
+
```

Comment:
As another alternative, `inotifywait` could be used to schedule an rsync run when files are changed. I have the following script in my path
```bash
#!/bin/sh
FORMAT=$(echo -e "\n\033[1;33m ==> %w%f written\033[0m")
"$@"
while inotifywait -qre close_write --format "$FORMAT" .
do
    "$@"
done
```

Reply:
so do you need to run this command every time you start working on your project ? 
it doesn't seems fool-proof.

Reply:
That is correct, but I find it reasonable to explicitly start a monitoring sync process like rsync. Anyway, that was just an additional idea.